### PR TITLE
v2.18.0

### PR DESCRIPTION
**Pull request by dbobrenko**
## Changes
  - InferenceTask switched from seeded determinism to logits.
  - Fix HF data sampling.
  - Fix DDGS web search.
  - Timeout adjustments.

### CODE REVIEW

**Review by bkb2135 — APPROVED** [review]
Looks Great! Good work!
